store/assets/currencies.go

```diff
@@ -0,0 +1,39 @@
+package assets
+
+import (
+	"fmt"
+	"math/big"
+)
+
+// Link contains a field to represent the smallest units of LINK
+type Link struct {
+	wei *big.Int
```

Comment:
This looks great. I'm loving how the compiler will prevent us from cross adding funds. In the interest of being thorough, I wanted to show you how `go-ethereum` handles new types: https://github.com/ethereum/go-ethereum/blob/master/common/hexutil/json.go#L133

```golang
type Big big.Int
```

They create a new type entirely (not type aliasing), rather than having a field. I think it's an approach worth exploring and it shouldn't be too far from what you have. We actually do this ourselves for `Cron` and a handful of types: https://github.com/smartcontractkit/chainlink/blob/master/store/models/common.go#L263

Reply:
@dimroc pushed a new commit with the type alias implementation. Was that what you were thinking?